lib/handleRequest.js

```diff
@@ -86,7 +82,17 @@ function preValidationCallback (err, request, reply) {
     return
   }
 
-  const validationErr = validateSchema(reply[kRouteContext], request)
+  let validationErr
+  try {
+    // TODO: Consider using Promise.try when Node.js 20 support is dropped (requires Node.js 22.10+)
+    validationErr = validateSchema(reply[kRouteContext], request)
```

Comment:
```suggestion
    validationErr = validateSchema(reply[kRouteContext], request)
```

@gurgunday note that `Promise.try` is a _bad idea_ here, as it would always add a microtick, which really we should not be doing if it's not needed (the vast majority of the case validation should not do any I/O).